src/platform.ts

```diff
@@ -78,6 +94,29 @@ export class WeatherFlowTempestPlatform implements DynamicPlatformPlugin {
           // Initialize sensors after first API response.
           this.discoverDevices();
 
+          this.log.debug ('discoverDevices completed');
+
+          // Remove cached sensors that are no longer required.
+          this.removeDevices();
+
+          this.log.debug ('removeDevices completed');
+
+          // Determine Tempest device_id & initial battery level
+          this.tempestApi.getTempestDeviceId().then( (device_id: number) => {
+            this.tempest_device_id = device_id;
```

Comment:
Thanks for moving this out to the Platform object.

Reply:
Made more sense in Platform object. Always something to learn!